geode-docs/developing/transactions/directed_example.html.md.erb

```diff
@@ -0,0 +1,256 @@
+an exception is thrown, and this example tries again.
+
+```
+// inputs needed for this transaction; shown as variables for simplicity
+final String customer = "Customer1";
+final Integer purchase = Integer.valueOf(1000);
+
+// region set up shown to promote understanding
+Cache cache = new CacheFactory().create();
+Pool pool = PoolManager.createFactory()
+           .addLocator("localhost", LOCATOR_PORT)
+           .create("pool-name");
+Region<String, Integer> cash =
+           cache.createClientRegionFactory(ClientRegionShortcut.PROXY)
+           .setPoolName(pool.getName())
+           .create("cash");
+Region<String, Integer> trades = 
+           cache.createClientRegionFactory(ClientRegionShortcut.PROXY)
+           .setPoolName(pool.getName())
+           .create("trades");
+
+// transaction code
+CacheTransactionManager txmgr = cache.getCacheTransactionManager();
+boolean retryTransaction = false;
+boolean transactionSucceeded = false;
+do {
+  try {
+    txmgr.begin();
+
+    // Subtract out the cost of the trade for this customer's balance
+    Integer cashBalance = cash.get(customer);
+    Integer newBalance = 
+       Integer.valueOf((cashBalance != null ? cashBalance : 0) - purchase);
+    cash.put(customer, newBalance);
+
+    // Add in the cost of the trade for this customer
+    Integer tradeBalance = trades.get(customer);
+    newBalance = 
+       Integer.valueOf((tradeBalance != null ? tradeBalance : 0) + purchase);
+    trades.put(customer, newBalance);
+
+  } catch (Exception exception) {
+    // Let the caller handle any exceptions caused when doing these region
+    // operations. Clean up the transactional state by doing a rollback.
+    // Examples of some exceptions: the data is not colocated, a rebalance
+    // interfered with the transaction, or the server is gone.
+    txmgr.rollback();
+    throw exception;
+  }
+
+  try {
+    txmgr.commit();
+    retryTransaction = false;
+    transactionSucceeded = true;
+  } 
+  catch (CommitConflictException conflict) {
+    // entry value changed causing a conflict for this customer, so try again
+    retryTransaction = true;
+  }
+  // All other exceptions that happend upon commit() are not handled, so
+  // the caller will have to handle those other exceptions. This code does
+  // not need a rollback(), since the exception happened on commit().
+
+} while (retryTransaction);
+```
+
+Design transactions such that any get operations are within the transaction.
+This causes those entries to be part of the transactional state,
+which is desired such that intersecting transactions can be detected
+and signal commit conficts.
+
+Note that variables `cashBalance` and `tradeBalance` are
+references,
+because the `Region.get` operation returns a reference.
+Read [Region Operations Return References](design_considerations.html#copy-on-read-transactions)
+for details and how to work around the
+implications of a reference as a return value.
```

Comment:
Technically, you can only *possibly* get a reference if you are running on the server side and you have a specific configuration. If you are using a client, you don't really have to worry about this.

We should probably just tell people to turn on copy-on-read if they are using transactions on the server side and leave it at that.